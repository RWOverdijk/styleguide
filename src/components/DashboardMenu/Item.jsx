import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './styles.scss';

const Item = ({ active, className, onClick, children, ...props }) => (
  <li
    className={classnames(
      'Van-DashboardMenu-item',
      { 'Van-DashboardMenu-item--active': active },
      className
    )}
    role="presentation"
    onClick={onClick}
    {...props}
  >
    {children}
  </li>
);

Item.defaultProps = {
  children: null,
  active: false,
  onClick: undefined,
  className: ''
};

Item.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
  active: PropTypes.bool
};

export default Item;
