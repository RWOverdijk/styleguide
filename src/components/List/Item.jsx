import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class Item extends PureComponent {
  render() {
    const { className, children, ...props } = this.props;

    return (
      <li className={classnames('Van-List-item', className)} {...props}>
        {children}
      </li>
    );
  }
}

Item.defaultProps = {
  className: '',
  children: ''
};

Item.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
};

export default Item;
