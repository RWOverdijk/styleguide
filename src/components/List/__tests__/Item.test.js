import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../Item';

describe("<Item>", () => {
  it("renders regular Item", () => {
    const tree = renderer.create(
      <Item>test</Item>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Item className="my-class">test</Item>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Item data-id="0">test</Item>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
