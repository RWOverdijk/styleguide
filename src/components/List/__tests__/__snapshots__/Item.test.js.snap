// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item> renders regular Item 1`] = `
<li
  className="Van-List-item"
>
  test
</li>
`;

exports[`<Item> renders with specific attributes 1`] = `
<li
  className="Van-List-item"
  data-id="0"
>
  test
</li>
`;

exports[`<Item> renders with specific className 1`] = `
<li
  className="Van-List-item my-class"
>
  test
</li>
`;
